darts/models/components/layer_norm_variants.py

```diff
@@ -54,3 +54,55 @@ def __init__(self, input_size, **kwargs):
 class LayerNorm(nn.LayerNorm):
     def __init__(self, input_size, **kwargs) -> None:
         super().__init__(input_size, **kwargs)
+
+
+class RINorm(nn.Module):
+    def __init__(self, input_dim: int, eps=1e-5, affine=True):
+        """Reversible Instance Normalization based on [1]
+
+        Parameters
+        ----------
+        input_dim
+            The dimension of the input axis being normalized
+        eps
+            The epsilon value for numerical stability
+        affine
+            Whether to apply an affine transformation after normalization
+
+        References
+        ----------
+        .. [1] Kim et al. "Reversible Instance Normalization for Accurate Time-Series Forecasting against
+                Distribution Shift" International Conference on Learning Representations (2022)
+        """
+
+        super().__init__()
+        self.input_dim = input_dim
+        self.eps = eps
+        self.affine = affine
+
+        if self.affine:
+            self.affine_weight = nn.Parameter(torch.ones(self.input_dim))
+            self.affine_bias = nn.Parameter(torch.zeros(self.input_dim))
+
+    def forward(self, x):
+        calc_dims = tuple(range(1, x.ndim - 1))
```

Comment:
I added some comments. I want to keep it as taking a torch.Tensor for now and not a specific data set. I may change this when it comes to integrating it with models since it may make covariate handling very gross in the models. 

When I add it to TiDE (and maybe NHiTS for testing) I will see how it looks and update it accordingly with another PR.